### PR TITLE
Fix nested object get and set not working in res files

### LINKED ISSUE #489: Nested object get and set not working in res files
Nested object set syntax is currently rejected with an `assert false` triggered at https://github.com/melange-re/melange/blob/f4ddebbc7755af5e983c4df2ac51498f286e1074/jscomp/frontend/ast_exp_apply.ml#L252

Input:
```rescript
obj["foo"]["bar"] = 42
```

Error:
```
dune build @melange
File "src/ui/.ui.objs/melange/_unknown_", line 1, characters 0-0:
melc: internal error, uncaught exception:
      File "jscomp/frontend/ast_exp_apply.ml", line 252, characters 17-23: Assertion failed
```

Additionally, there's a related bug in nested gets.

```Input:
x["foo"]["bar"]
```

Produces an incorrect compiled output:
```
x.bar
```

### PR DESCRIPTION
**Pull request by jchavarri**
Fixes #489.

### CODE REVIEW

**Comment by jchavarri** [comment]
@anmonteiro the fix is ready, but I think I need some help configuring dune so that rescript-syntax package is added as dep of this test. 😅

**Comment by anmonteiro** [comment]
`-p` is `only-packages`. How about `dune build runtest -p melange,rescript-syntax,reactjs-jsx-ppx`?

**Comment by jchavarri** [comment]
Yes, this works, see Makefile:


https://github.com/melange-re/melange/blob/f550ed977e67e35b8e5177365cdfb152d177fa09/Makefile#L23-L24


The problem happens when installing melange, because `melange.opam` has the following on `build`:

https://github.com/melange-re/melange/blob/f550ed977e67e35b8e5177365cdfb152d177fa09/melange.opam#L24-L31

**Comment by jchavarri** [comment]
@anmonteiro I think I got it to a point where everything works without any workarounds. There were a series of issues. I ended up creating a folder `test/rescript-syntax` and moving the new test of this package over there, to avoid having to use `applies_to`, which seems it's a bit buggy and cumbersome to use. I also started using `%{bin:xxx}` dependencies rather than `package` ones, which seems to work and doesn't lead into the issue you reported in https://github.com/ocaml/dune/pull/7220.

File: rescript-syntax/ppx_rescript_compat/ppx_rescript_compat.ml [hunk 1]
```diff
@@ -47,10 +47,11 @@ let expr_mapper (self : mapper) (expr : Parsetree.expression) =
             args;
       }
   | Pexp_send
-      ( ({ pexp_desc = Pexp_apply _ | Pexp_ident _; pexp_loc; _ } as subexpr),
+      ( ({ pexp_desc = Pexp_apply _ | Pexp_ident _ | Pexp_send _; pexp_loc; _ } as subexpr),
```
Comment (anmonteiro) [inline rescript-syntax/ppx_rescript_compat/ppx_rescript_compat.ml:50]: sure this fixes the case `((entry ## person) ## age)`, but does it fix for more `##`?

Comment (jchavarri) [reply rescript-syntax/ppx_rescript_compat/ppx_rescript_compat.ml:50]: it does, added another level of depth in https://github.com/melange-re/melange/pull/533/commits/2ca2df93982008272d58b47b4e2c9aa3fb6890cd.

**Comment by anmonteiro** [comment]
Thank you!

**Pull request merged**